specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,639 @@
+It contains cryptographic public-key material for one or more technologies.
+
+The VOA hierarchy organizes _signature verifiers_ by [os], [purpose], [context] and [technology], and stores them in a directory structure of the form `$os/$purpose/$context/$technology/`.
+
+Vendors provide a set of _signature verifier_ files for their OS, in the VOA hierarchy format.
+
+Users of a VOA hierarchy (such as package installation software) can pick the relevant set of _signature verifiers_ for the verification of a specific artifact based on their location in the hierarchy.
+
+VOA does not concern itself with the retrieval of additional _signature verifiers_ or the update of existing ones in the file hierarchy.
+
+### Load paths
+
+VOA defines a list of _load paths_ with descending priority for system mode and user mode.
+
+_Signature verifiers_ are retrieved from hierarchies in these directories to validate artifacts based on specific [load logic].
+This is referred to as _verifier lookup_.
+
+The existence of multiple load paths allows system administrators and users to define custom sets of _signature verifiers_ for their system and applications.
+
+#### System mode
+
+The priority of load paths in system mode follows the definition of ["Storage Directories and Overrides" in the Configuration Files Sepcification], which is also reflected in the following list of paths:
+
+- `/etc/voa/`
+- `/run/voa/`
+- `/usr/local/share/voa/`
+- `/usr/share/voa/`
+
+#### User mode
+
+The priority of load paths in user mode mirrors that of the system mode.
+Refer to the [XDG Base Directory Specification] for details on default values.
+
+- `$XDG_CONFIG_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_CONFIG_DIRS`
+- `$XDG_RUNTIME_DIR/voa/`
+- `$XDG_DATA_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_DATA_DIRS`
+
+### Symlinking
+
+Load paths are constrained to self-contained locations on a host as they provide vital data for the integrity and verification of all components on a system.
+However, symlinks can be used in the VOA hierarchy to point to files or directories below one of the [load paths] in descending priority.
+Symlinks to files or directories below ephemeral load paths (i.e. `/run/voa/` and `$XDG_RUNTIME_DIR/voa/`) are prohibited, as they would lead to dangling references.
+
+As an example, symlinks to files below the same load path or to another load path with lower priority may be used to deduplicate the use of a single _signature verifier_ for multiple use-cases.
+Additionally, using symlinks allows to automatically keep _signature verifiers_ in sync with canonical upstream data.
+
+Symlinking to files or directories external to the load paths is prohibited.
+VOA implementations must not consider symlinks to files outside of the specified load paths and should raise a warning if such symlinks are encountered.
```

Comment:
you probably want to say here that symlinks must be relative, may not be absolute.

(that's kinda handy so that people can bind mount a voa tree into a container and it works)

Reply:
Thanks for clarifying this! We have adapted and simplified the symlinking section accordingly.